test/e2e/network_segmentation_default_network_annotation.go

```diff
@@ -103,4 +103,92 @@ var _ = Describe("Network Segmentation: Default network multus annotation", func
 			mac: "02:A1:B2:C3:D4:E5",
 		}),
 	)
+
+	Context("ValidatingAdmissionPolicy protection", func() {
+		It("should prevent adding, modifying and removing the default-network annotation on existing pods", func() {
+			if !isPreConfiguredUdnAddressesEnabled() {
+				Skip("ENABLE_PRE_CONF_UDN_ADDR not configured")
+			}
+
+			namespace, err := f.CreateNamespace(context.TODO(), f.BaseName, map[string]string{
+				"e2e-framework":           f.BaseName,
+				RequiredUDNNamespaceLabel: "",
+			})
+			Expect(err).NotTo(HaveOccurred(), "Should create namespace for test")
+			f.Namespace = namespace
+
+			udnClient, err := udnclientset.NewForConfig(f.ClientConfig())
+			Expect(err).NotTo(HaveOccurred(), "Should create UDN client")
+
+			// Create a UserDefinedNetwork for the test
+			udn := &udnv1.UserDefinedNetwork{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-network",
+					Namespace: f.Namespace.Name,
+				},
+				Spec: udnv1.UserDefinedNetworkSpec{
+					Topology: udnv1.NetworkTopologyLayer2,
+					Layer2: &udnv1.Layer2Config{
+						Role: udnv1.NetworkRolePrimary,
+						Subnets: filterDualStackCIDRs(f.ClientSet, []udnv1.CIDR{
+							"103.0.0.0/16",
+							"2014:100:200::0/60",
+						}),
+					},
+				},
+			}
+
+			By("Creating a UserDefinedNetwork")
+			udn, err = udnClient.K8sV1().UserDefinedNetworks(f.Namespace.Name).Create(context.TODO(), udn, metav1.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred(), "Should create UserDefinedNetwork")
+			Eventually(userDefinedNetworkReadyFunc(f.DynamicClient, udn.Namespace, udn.Name), 5*time.Second, time.Second).Should(Succeed())
```

Comment:
It seems the UDN is not attached to the pods, is it intentional? if not I think we can drop it.

Reply:
It is, this is a primary UDN that is automatically attached to the pods in the namespace.
Did you mean anything else?

Reply:
I didnt realize its primary, sorry for the noise.